flask_mongorest/resources.py

```diff
@@ -344,6 +357,14 @@ def value_for_field(self, obj, field):
         raise UnknownFieldError
 
     def validate_request(self, obj=None):
+        if (request.method == 'PUT' and obj) or request.method == 'POST':
```

Comment:
These conditions are always met - we only use `validate_request` in `ResourceView#post` and `ResourceView#put` (where for PUT, `obj` is never `None`).